server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -818,7 +818,9 @@ public void publish(ClusterChangedEvent clusterChangedEvent, ActionListener<Void
                 // there is no equals on cluster state, so we just serialize it to XContent and compare JSON representation
                 assert clusterChangedEvent.previousState() == coordinationState.get().getLastAcceptedState() ||
                     Strings.toString(clusterChangedEvent.previousState()).equals(
-                        Strings.toString(clusterStateWithNoMasterBlock(coordinationState.get().getLastAcceptedState())));
+                        Strings.toString(clusterStateWithNoMasterBlock(coordinationState.get().getLastAcceptedState())))
+                    : clusterChangedEvent.previousState() + " vs "
```

Comment:
maybe print the toString(...) variants here because it's those we check for equality?